features/0434-outofband/README.md

```diff
@@ -130,7 +130,57 @@ If only the `request~attach` item is included, no new connection is expected to
 
 If both the `handshake_protocols` and `request~attach` items are included in the message, the receiver should first establish a connection and then respond (using that connection) to one of the messages in the `request~attach` message. If a connection already exists between the parties, the receiver may respond immediately to the `request-attach` message using the established connection.
 
-While the _receiver_ is expected to respond with an initiating message from a `handshake_protocols` or `request~attach` item using an offered service, the receiver may be able to respond by reusing an existing connection. Specifically, if a connection they have was created from an out-of-band `invitation` from the same public DID of a new `invitation` message, the receiver **SHOULD** use the existing connection in responding to the `invitation`. If a selected `handshake_protocols` item is used by the receiver, it may have a specific "reuse this connection" message type.
+### Reuse Messages
+
+While the _receiver_ is expected to respond with an initiating message from a `handshake_protocols` or `request~attach` item using an offered service, the receiver may be able to respond by reusing an existing connection. Specifically, if a connection they have was created from an out-of-band `invitation` from the same public DID of a new `invitation` message, the receiver **MAY** use the existing connection in responding to the `invitation`. This is accomplished using the `reuse` and `reuse-accepted` messages.
+
+#### Reuse: `https://didcomm.org/out-of-band/%VER/reuse`
+
+```jsonc
+{
+  "@type": "https://didcomm.org/out-of-band/%VER/reuse",
+  "@id": "<id>",
+  "~thread": {
+      "thid": "<same as @id>",
+      "pthid": "<The @id of the Out-of-Band invitation>"
+  }
+}
+```
+
+The items in the message are:
+
+- `@type` - the DIDComm message type
+- `@id` - the unique ID of the message. 
+- `pthid` - the @id of the invitation message. This provides the context link for the _inviter_ to prompt additional protocol interactions.
+
+Sending or receiving this message does not change the state of the existing connection.
+
+When the _inviter_ receives the `reuse` message, they **MUST** respond with a `reuse-accepted` message to notify that _invitee_ that the request to reuse the existing connection is successful.
+
+#### Reuse Accepted:  `https://didcomm.org/out-of-band/%VER/reuse-accepted`
```

Comment:
I would argue this is not needed.  The "done" (I think that's the name) is to complete the handshake. If the connection exists and is active, is there a need to have another confirmation message?

There is the scenario to consider if there being an existing connection, the invitation receiver sending the reuse, and the invitation sender having deleted the connection. However, I don't think that case is relevant as either:

- the invitation sender is using the public DID for the connection and understands the reuse message and can continue, or
- the invitation sender is using a peer DID for the connection, in which case the "reuse" message cannot be opened, and may not even get to the sender.

Reply:
Given the discussion in today's Aries WG call (2020-06-17), we are going to leave it here.